### PR TITLE
[LLVM] BasicTTIImpl allow unknown type during legality checking

### PR DESCRIPTION
**Pull request by zyx-billy**
Make BasicTTIImplBase's `isTypeLegal` check handle unknown types. Current behavior is aborting.

Motivated by a use case in SimplifyCFG, where `isTypeLegal` is called on a struct type and dies, when it could be treated as illegal and skipped. In general it could make sense for unknown types to be allowed, and by default just considered not legal, but the behavior can of course be overriden. Please LMK if this is too lenient for other use cases (we can consider parameterizing it too).

### CODE REVIEW

**Review by Mogball — APPROVED** [review]

**Comment by Mogball** [comment]
LGTM but please get a review from someone else

File: llvm/test/Transforms/SimplifyCFG/X86/switch_to_lookup_table.ll
Comment (khei4) [inline llvm/test/Transforms/SimplifyCFG/X86/switch_to_lookup_table.ll:2076]: You should use [llvm/utils/update_test_checks.py](https://github.com/llvm/llvm-project/blob/main/llvm/utils/update_test_checks.py), for transform tests especially for which have `; NOTE: Assertions have been autogenerated by utils/update_test_checks.py` for L1.

Comment (zyx-billy) [reply llvm/test/Transforms/SimplifyCFG/X86/switch_to_lookup_table.ll:2076]: oh I see. Thanks for the tip!

**Comment by khei4** [comment]
LGTM. But I think it's better to be reviewed by recently active committers. 
@nikic @dtcxzyw @DianQK

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**
